### PR TITLE
Allow next-css to accept css-loader options

### PR DESCRIPTION
**Pull request by thekevinscott**
This PR accompanies https://github.com/zeit/next-plugins/issues/15.

I can't find a way to pass options to the webpack css-loader, so this PR enables that. It should also work with `next-sass` (and other plugins relying on `next-css`).

In this update, `cssModules` accepts either a string or an object. 

If a **string**, behavior is as before (`modules` is set to the value of `cssModules`).

If an **object**, the `options` object is a join of default options and the user-specified `cssModules` object (with user-specified taking precedence).

Questions:

1) Would it be clearer to have a dedicated options argument, like `cssModulesOptions`? (I tried to make the API change as minimal as possible but maybe it's clearer as an additional argument).
2) In the example below, if a user neglects to provide a `modules` argument in the `cssModules` object they are passing, `modules` will take on an incorrect value of `[Object object]`. We could only set the value of `modules` if `cssModules` is a string. I elected to make the code more readable by omitting this, but maybe it's worth considering

### CODE REVIEW

**Comment by gihrig** [comment]
@thekevinscott I tried out your update and it works flawlessly 💯 

Re: Questions (above)
> 1. Would it be clearer to have a dedicated options argument...

The options argument and the way you documented it in the readme was perfect for me. 

It would have been nice to have a more complete example, showing the use of the modified css. e.g.

```javascript
import styles from "./style.css"

const Component = props => {
  return (
    <div className={styles.backdrop}>
```
Or just refer to [CSS Modules — Local Scope](https://github.com/css-modules/css-modules/blob/master/docs/local-scope.md#css-modules--local-scope) docs.

>2. In the example below,...

I don't see an example. Are you referring to the readme?

> 2. ... if a user neglects to provide a modules argument in the cssModules object they are passing, modules will take on an incorrect value of [Object object]. ...

I didn't see a problem with this.

When I leave out the `modules: 1,` argument (see below) everything works as expected. If I just leave off the value for modules `modules:,`, a clearly defined syntax error is thrown. So I think it's fine just the way it is.

```bash
next.config.js:7
    modules:,
            ^
SyntaxError: Unexpected token ,
```

```javascript
module.exports = withCSS({
  cssModules: {
    importLoaders: 1,
    localIdentName: '[name]_[local]-[hash:base64:4]',
  }
})
```

From my testing, this PR is ready to merge (with a minor tweak to the docs) 😄

**Comment by thekevinscott** [comment]
Sweet! I'll make the doc changes and push an update momentarily.

**Comment by thekevinscott** [comment]
Updated!

**Comment by gihrig** [comment]
Love those doc updates! 💯 

Let's land this ✨

File: packages/next-css/css-loader-config.js [hunk 1]
```diff
@@ -5,15 +5,25 @@ module.exports = (
...
 ) => {
+  const defaultOptions = {
+    minimize: !dev,
+    sourceMap: dev,
+    importLoaders: 1,
+    modules: cssModules,
+  };
+
+  let options = defaultOptions;
+  if (typeof cssModules === "object") {
```
Comment (timneutkens) [inline packages/next-css/css-loader-config.js]: Lets make it a new key `cssLoaderOptions` instead of overriding `cssModules`

**Comment by timneutkens** [comment]
Great work @thekevinscott! Just had one point, after that I'll merge 🙏

**Comment by thekevinscott** [comment]
Thanks @timneutkens !

**Comment by sahalsajjad** [comment]
@thekevinscott Thanks for your update. #

Have you added **camelCase** option ? It works for class names without dashes, however it returns an empty string when there is a dash in the css selector (eg: _.navbar-bg_). Could you please add the camelCase option of css-loader.

Many of our legacy css code have class selectors with dashes.
Thanks in advance. :tada:

**Comment by timneutkens** [comment]
This is not released yet.

**Pull request merged**